core/templates/services/playthrough-issues-backend-api.service.ts

```diff
@@ -15,87 +15,117 @@
+    } else {
+      // TODO(#7165): Change `any` to a type describing the dict.
+      return this.httpClient.get(
+        this.getFullIssuesUrl(explorationId), {
+          params: { exp_version: explorationVersion },
+          observe: 'response'
+        }).toPromise().then(
+        (response: HttpResponse<any[]>) => {
+          let unresolvedIssueBackendDicts = response.body;
+          this.cachedIssues = unresolvedIssueBackendDicts.map(
+            this.playthroughIssueObjectFactory.createFromBackendDict);
+          return this.cachedIssues;
         });
-    };
-    return {
-      fetchIssues: function(explorationId, explorationVersion) {
-        if (cachedIssues !== null) {
-          return $q.resolve(cachedIssues);
-        } else {
-          return $http.get(getFullIssuesUrl(explorationId), {
-            params: {
-              exp_version: explorationVersion
-            }
-          }).then(function(response) {
-            var unresolvedIssueBackendDicts = response.data;
-            cachedIssues = unresolvedIssueBackendDicts.map(
-              PlaythroughIssueObjectFactory.createFromBackendDict);
-            return cachedIssues;
-          });
-        }
-      },
-      fetchPlaythrough: function(expId, playthroughId) {
-        return $http.get(getFullPlaythroughUrl(expId, playthroughId)).then(
-          function(response) {
-            var playthroughBackendDict = response.data;
-            return PlaythroughObjectFactory.createFromBackendDict(
-              playthroughBackendDict);
-          });
-      },
-      resolveIssue: function(issueToResolve, expId, expVersion) {
-        return $http.post(getFullResolveIssueUrl(expId), {
-          exp_issue_dict: issueToResolve.toBackendDict(),
-          exp_version: expVersion
-        }).then(function() {
-          var issueIndex = cachedIssues !== null ?
-            cachedIssues.findIndex(function(issue) {
-              return angular.equals(issue, issueToResolve);
-            }) : -1;
-          if (issueIndex === -1) {
-            var invalidIssueError = new Error(
-              'An issue which was not fetched from the backend has been ' +
-              'resolved');
-            return $q.reject(invalidIssueError);
-          } else {
-            cachedIssues.splice(issueIndex, 1);
-          }
+    }
+  }
+
+  fetchPlaythrough(
+      explorationId: string, playthroughId: string): Promise<PlaythroughIssue> {
+    // TODO(#7165): Change `any` to a type describing the dict.
+    return this.httpClient.get(
+      this.getFullPlaythroughUrl(explorationId, playthroughId), {
+        observe: 'response'
+      }).toPromise().then((response: HttpResponse<any>) => {
+      let playthroughBackendDict = response.body;
+      return this.playthroughIssueObjectFactory.createFromBackendDict(
+        playthroughBackendDict);
+    });
+  }
+
+  resolveIssue(
+      issueToResolve: PlaythroughIssue, explorationId: string,
+      expVersion: number): Promise<any> {
```

Comment:
According to the success case we aren't actually returning anything, right? Seems to me that this should be changed to have a return type of `Promise<void>`.

Reply:
If i change it to void it throws this error
`error TS2322: Type 'Promise<void | Observable<never>>' is not assignable to type 'Promise<void>'.`

 ` Type 'void | Observable<never>' is not assignable to type 'void'.`

  `  Type 'Observable<never>' is not assignable to type 'void'.`

Reply:
Please add me onto that PR, I would like to know how this is solved, thanks